src/pybind/mgr/dashboard/frontend/src/app/shared/components/deletion-button/deletion-button.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Component, EventEmitter, Input, OnInit, Output, TemplateRef,
+  ViewChild
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { BsModalRef, BsModalService } from 'ngx-bootstrap';
+import { Observable } from 'rxjs/Observable';
+import { SubmitButtonComponent } from '../submit-button/submit-button.component';
+
+@Component({
+  selector: 'cd-deletion-button',
+  templateUrl: './deletion-button.component.html',
+  styleUrls: ['./deletion-button.component.scss']
+})
+export class DeletionButtonComponent implements OnInit {
+  @ViewChild(SubmitButtonComponent) submitButton: SubmitButtonComponent;
+  @Input() metaType: string;
+  @Input() pattern: string;
+  @Input() btnClasses = 'btn btn-sm btn-primary';
+  @Input() deletionObserver: () => Observable<any>;
+  @Output() toggleDeletion = new EventEmitter();
+  bsModalRef: BsModalRef;
+  deletionForm: FormGroup;
+  confirmation: FormControl;
+  delete: Function;
+
+  constructor(private modalService: BsModalService) {}
+
+  ngOnInit() {
+    this.confirmation = new FormControl('', {
+      validators: [
+        Validators.required,
+        Validators.pattern(this.pattern)
+      ],
+      updateOn: 'blur'
+    });
+    this.deletionForm = new FormGroup({
+      confirmation: this.confirmation
+    });
+  }
+
+  showModal(template: TemplateRef<any>) {
+    this.deletionForm.reset();
+    this.bsModalRef = this.modalService.show(template);
+    this.delete = () => {
+      this.submitButton.submit();
+    };
+  }
+
+  invalidControl (control: FormControl, error?: string) {
+    return control.dirty && control.invalid && (error ? control.errors[error] : true);
```

Comment:
We should also check if the form is in the submitted state. something like `(formDir.submitted || control.dirty) && ..`

Reply:
Included